docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -254,7 +249,8 @@ public Node call() throws Exception {
                                 // TODO: record the output somewhere
                                 DockerSlave slave = null;
                                 try {
-                                    slave = t.provision(new StreamTaskListener(System.out));
+//                                    slave = t.provision(new StreamTaskListener(System.out));
+                                    slave = provision(t);
```

Comment:
Look at the evilness below here:

```
jenkins.addNode(slave);
```

NOOOOOOOOOOOOOOOOOO! Leave that up to NodeProvisioner as per [javadoc](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/slaves/Cloud.java#L145) and the baseline version of Jenkins is well well at least 1.503